client/app/components/dynamic-parameters/DateRangeParameter.jsx

```diff
@@ -0,0 +1,115 @@
+  { name: 'Last month', value: 'd_last_month', label: () => getDynamicDateRange('d_last_month').value()[0].format('MMMM') },
+  { name: 'Last year', value: 'd_last_year', label: () => getDynamicDateRange('d_last_year').value()[0].format('YYYY') },
+  { name: 'Last 7 days',
+    value: 'd_last_7_days',
+    label: () => getDynamicDateRange('d_last_7_days').value()[0].format('MMM D') + ' - Today' },
+];
+
+const DYNAMIC_DATETIME_OPTIONS = [
+  { name: 'Today',
+    value: 'd_today',
+    label: () => getDynamicDateRange('d_today').value()[0].format('MMM D') },
+  { name: 'Yesterday',
+    value: 'd_yesterday',
+    label: () => getDynamicDateRange('d_yesterday').value()[0].format('MMM D') },
+  ...DYNAMIC_DATE_OPTIONS,
+];
+
+function isValidDateRangeValue(value) {
+  return isArray(value) && value.length === 2 && moment.isMoment(value[0]) && moment.isMoment(value[1]);
+}
+
+class DateRangeParameter extends React.Component {
+  static propTypes = {
+    type: PropTypes.string,
+    className: PropTypes.string,
+    value: PropTypes.any, // eslint-disable-line react/forbid-prop-types
+    parameter: PropTypes.any, // eslint-disable-line react/forbid-prop-types
+    onSelect: PropTypes.func,
+  };
+
+  static defaultProps = {
+    type: '',
+    className: '',
+    value: null,
+    parameter: null,
+    onSelect: () => {},
+  };
+
+  onDynamicValueSelect = (dynamicValue) => {
+    const { onSelect, parameter } = this.props;
+    if (dynamicValue === 'static') {
+      onSelect([moment(parameter.getValue().start), moment(parameter.getValue().end)]);
+    } else {
+      onSelect(dynamicValue.value);
+    }
+  };
+
+  render() {
+    const { type, value, parameter, onSelect, className } = this.props;
+    const isDateTimeRange = includes(type, 'datetime-range');
+    const hasDynamicValue = isDynamicDateRange(value);
+    const options = isDateTimeRange ? DYNAMIC_DATETIME_OPTIONS : DYNAMIC_DATE_OPTIONS;
+
+    const additionalAttributes = {};
+
+    let DateRangeComponent = DateRangeInput;
+    if (isDateTimeRange) {
+      DateRangeComponent = DateTimeRangeInput;
+      if (includes(type, 'with-seconds')) {
+        additionalAttributes.withSeconds = true;
+      }
+    }
+
+    if (isValidDateRangeValue(value) || value === null) {
+      additionalAttributes.value = value;
+    }
+
+    if (hasDynamicValue) {
+      const dynamicDateRange = getDynamicDateRange(value);
+      additionalAttributes.placeholder = [dynamicDateRange && dynamicDateRange.name];
+      additionalAttributes.value = null;
+    }
+
+    return (
+      <DateRangeComponent
+        className={classNames('redash-datepicker', { 'dynamic-value hide-end-value': hasDynamicValue }, className)}
+        onSelect={onSelect}
+        suffixIcon={(
+          <DynamicButton
+            options={options}
+            selectedDynamicValue={(hasDynamicValue && isString(parameter.value)) ? parameter.value : null}
```

Comment:
Doesn't seem to work in date range:

<img width="245" alt="Screen Shot 2019-07-20 at 18 34 48" src="https://user-images.githubusercontent.com/486954/61580674-271b1b80-ab1d-11e9-9578-a561f62515d1.png">

Reply:
Actually, what's happening is that the selected value is `param.value` but should be the "pending" value if there is one.

Reply:
(I just realized the arrow is pointing to the wrong option but you know what I mean 😄)

Reply:
After the merge with Apply All `parameter.value` and `value` became a different thing ^^, so I had to change a lot on the implementation, this was a missing point :grimacing:.